### DISCUSSION TITLE
Use gpt-engineer to improve gpt-engineer – open source contribs wanted

### DISCUSSION THREAD

**Discussion by AntonOsika** [AntonOsika/gpt-engineer#981] [Ideas]
Anyone want to help contributing to codegen opensource potentially using codegen?

Just opened this enhancement issue.

Should be possible to fix with gpt-engineer itself, using -i flag:

https://github.com/gpt-engineer-org/gpt-engineer/issues/980

(one might need to grep codebase for PrepromptsHolder, and pipe in all matching filenames to the file_selector flow)

**Comment by 6rz6** [comment]
gladly, added the Ollama Multi threaded agent UI, working on real time resource management when running multiple agents with different LLMs simultaneously  and [integrated into the main branch on v0.28](https://github.com/gpt-engineer-org/gpt-engineer/pull/1098/files#r1550253557) 

[![pre-commit](https://github.com/gpt-engineer-org/gpt-engineer/actions/workflows/pre-commit.yaml/badge.svg)](https://github.com/gpt-engineer-org/gpt-engineer/actions/workflows/pre-commit.yaml)
